### PR TITLE
Add a question when a user hasn't selected a log type

### PR DESCRIPTION
**Pull request by robindevitt**
When a user requests logs with `composer server logs` and doesn't pass a service they are presented with an error that isn't clear.

The user will now be presented with options of the different services that they can select from. The default service will be `php` and the remaining options are presented in alphabetical order.

Fixes https://github.com/humanmade/altis-local-server/issues/216

### CODE REVIEW

File: inc/composer/class-command.php [hunk 1]
```diff
@@ -492,7 +493,30 @@ protected function status( InputInterface $input, OutputInterface $output ) {
 	 * @return int
 	 */
 	protected function logs( InputInterface $input, OutputInterface $output ) {
-		$log = $input->getArgument( 'options' )[0];
+		var_dump( 'start' );
```
Comment (roborourke) [inline inc/composer/class-command.php:496]: Debug code?

File: inc/composer/class-command.php [hunk 2]
```diff
@@ -492,7 +493,30 @@ protected function status( InputInterface $input, OutputInterface $output ) {
...
+			$question = new ChoiceQuestion(
+				'Please select your prefered service (defaults to php)',
+				[
+					'php',
+					'cavalcade',
+					'db',
+					'elasticsearch',
+					'nginx',
+					'redis',
+					's3',
```
Comment (roborourke) [inline inc/composer/class-command.php:508]:
```suggestion
					's3',
					'tachyon',
```

File: inc/composer/class-command.php [hunk 3]
```diff
@@ -492,7 +493,30 @@ protected function status( InputInterface $input, OutputInterface $output ) {
 	 * @return int
 	 */
 	protected function logs( InputInterface $input, OutputInterface $output ) {
-		$log = $input->getArgument( 'options' )[0];
+		var_dump( 'start' );
+		if( !isset( $input->getArgument( 'options' )[0] ) ){
+			$helper = $this->getHelper( 'question' );
+			$question = new ChoiceQuestion(
+				'Please select your prefered service (defaults to php)',
```
Comment (roborourke) [inline inc/composer/class-command.php:500]: Some slight wording changes here:
```suggestion
				'Please select a service (defaults to php)',
```

File: inc/composer/class-command.php [hunk 4]
```diff
@@ -492,7 +493,30 @@ protected function status( InputInterface $input, OutputInterface $output ) {
...
+					'db',
+					'elasticsearch',
+					'nginx',
+					'redis',
+					's3',
+					'xray',
+				],
+				0
+			);
+			$question->setErrorMessage( 'Selected service %s is invalid, please select again!' );
```
Comment (roborourke) [inline inc/composer/class-command.php:513]:
```suggestion
			$question->setErrorMessage( '%s is not a recognised service, please select again!' );
```

File: inc/composer/class-command.php [hunk 5]
```diff
@@ -492,7 +493,30 @@ protected function status( InputInterface $input, OutputInterface $output ) {
...
+			$service = $helper->ask( $input, $output, $question );
+			$output->writeln( 'You have selected: '.$service );
```
Comment (roborourke) [inline inc/composer/class-command.php:515]:
```suggestion
			$output->writeln( sprintf( '<comment>Fetching %s logs...</>', $service ) );
```

**Review by roborourke — CHANGES_REQUESTED** [review]
Couple of wording suggestions, and pay careful attention to detail with code style - there are a bunch of linter errors to fix up here first.

File: inc/composer/class-command.php (hunk 2 above)
Comment (robindevitt) [reply inc/composer/class-command.php:508]: Should Tachyon be added to the docs too? As this was what I used for the source of the list https://docs.altis-dxp.com/local-server/viewing-logs/

Comment (roborourke) [reply inc/composer/class-command.php:508]: Ah right now I remember, we don’t run tachyon in debug mode locally right now so there are no logs really. Leave it out for now

**Comment by robindevitt** [comment]
Hey @roborourke 
Thank you for the review, I have pushed the changed up.

**Review by roborourke — APPROVED** [review]

**Comment by roborourke** [comment]
Good stuff!

**Pull request merged**